ast/src/parsed/mod.rs

```diff
@@ -23,7 +23,8 @@ pub enum PilStatement<T> {
     PublicDeclaration(
         usize,
         String,
-        NamespacedPolynomialReference<T>,
+        NamespacedPolynomialReference,
+        Option<Expression<T>>,
         Expression<T>,
```

Comment:
This enum looks quite obscure now, a comment explaining what the arguments are would help.

Reply:
I can extract it into a struct.

Reply:
Although the parsed enums all don't have structs to make it easier with the grammar. I'll add some comments...

Reply:
I mean in the end this should be an Expresison, to be honest, and then checked by the condenser that it has the right format.

Reply:
Yeah, let me change this a bit.